### PR TITLE
Query and update against aliases field

### LINKED ISSUE #37: Query and update against aliases field in azul-index
It is not the desired behavior of this DOS to present discoverabiity metadata, since it may present controlled access data. It currently allows one to filter on any key (provided it is a string) in the azul-index.

To close this, change the behavior of alias filtering to only make requests against the alias field, like so:

```
{
  "query": {
    "match": {"aliases.keyword": "dg.4503/bc4bdd84-9c6e-4321-97b5-c4c9893f096b"}
  }
}
```

The update should also be change to modify just the aliases field of the index.

Update tests accordingly.

Thanks @mikebaumann

┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/DAL-27)
┆Project Name: dos-azul-lambda
┆Issue Number: DAL-27
┆Epic: DAL One point ohh

### LINKED ISSUE #64: Remove alias splitting and protected key logic
┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/DAL-48)
┆Project Name: dos-azul-lambda
┆Issue Number: DAL-48

### PR DESCRIPTION
**Pull request by natanlao**
Closes #37.
Closes #64.

### CODE REVIEW

**Comment by natanlao** [comment]
Sorry, I spammed the issues a little with commit messages... I had some problems with merge conflicts but it looks like everything is okay now. (As an aside, I successfully used `cherry-pick` to solve my problem which made me feel like an absolute pro.)

File: tests/test_app.py [hunk 1]
```diff
@@ -142,15 +142,11 @@ def test_update_unauthenticated(self):
...
         """
-        Helper test that demonstrates updating a data object with
-        a given alias.
-
-        :param bool expect_fail: expect that updating the alias will fail, and
-                                 stop once it does. This exists only for
-                                :meth:`test_update_protected_key`
+        Demonstrates updating a data object with a given alias.
         """
+        alias = 'daltest:' + str(uuid.uuid1())
```
Comment (david4096) [inline tests/test_app.py]: nice

File: tests/test_app.py [hunk 2]
```diff
@@ -142,15 +142,22 @@ def test_update_unauthenticated(self):
...
+    def test_nonexist_alias(self):
         """
-        Helper test that demonstrates updating a data object with
-        a given alias.
+        Test to ensure that looking up a nonexistent alias returns an
+        empty list.
+        """
+        alias = str(uuid.uuid1())
+        body = self.make_request('GET', '/ga4gh/dos/v1/dataobjects?alias=' + alias)
+        self.assertEqual(len(body['data_objects']), 0)
```
Comment (david4096) [inline tests/test_app.py]: Great!

**Comment by david4096** [comment]
Thanks!

**Pull request merged**